src/js/modules/supervisors/Repository.ts

```diff
@@ -87,31 +90,68 @@ class Repository {
     return this.handles.size;
   }
 
+  createEmptyProcessBatchReplay(
+    handle: Handle,
+    requestItem: ProcessBatchRequestItem
+  ): ProcessBatchReplyItem {
+    return {
+      coprocessorId: BigInt(handle.coprocessor.globalId),
+      ntp: {
+        ...requestItem.ntp,
+        topic: `${requestItem.ntp.topic}`,
+      },
+      resultRecordBatch: [],
+    };
+  }
+
   transformMapResultToArray(
     requestItem: ProcessBatchRequestItem,
     handle: Handle,
     recordBatch: RecordBatch,
     resultRecordBatch: Map<string, RecordBatch>
   ): ProcessBatchReplyItem[] {
     const results: ProcessBatchReplyItem[] = [];
-    for (const [key, value] of resultRecordBatch) {
-      value.records = value.records.map((record) => {
-        record.length = calculateRecordLength(record);
-        record.valueLen = record.value.length;
-        return record;
-      });
-      value.header.sizeBytes = calculateRecordBatchSize(value.records);
-      value.header.term = recordBatch.header.term;
-      results.push({
-        coprocessorId: BigInt(handle.coprocessor.globalId),
-        ntp: {
-          ...requestItem.ntp,
-          topic: `${requestItem.ntp.topic}.$${key}$`,
-        },
-        resultRecordBatch: [value],
-      });
-      return results;
+    if (resultRecordBatch === undefined) {
```

Comment:
@senior7515 @graphcareful @0x5d  what do you think about it? this condition handles when the apply function returns a Promise<undefined>, I propose to respond to redpanda with empty processBatchRequest

processBatchRequest* same NTP that origin request and any record batch

Reply:
Not sure we can either crash or implicitly cast null to an empty map in which we will be filtering the batch. However i'd favor a more explicit type of behavior. My opinion is to just crash if the apply() function doesn't return the exact type we expect.

Reply:
the validation for an empty map is next to this validation, but I'll add a crash if the apply function returns any other type different from Map

Reply:
I think undefined should crash. 

`Promise<Optional<RecordBatch>>` should be valid.

Reply:
`Promise<Optional<RecordBatch>>` I don't understand how I can process that type, maybe `Promise<Optional<Map<string,RecordBatch>>>`

if is `Promise<Optional<Map<string,RecordBatch>>>` we can handle de undefined like a `None` in that `Optional`

Reply:
I'll implement, follow the approach if someone returns any other type different from Promise<Map>, that coprocessor is going to be disabled and log the error